src/plugins/graph/utilities/graph-utils.ts

```diff
@@ -140,37 +141,203 @@ export interface IMatchCirclesProps {
+      dotsElement, pointRadius, pointColor, pointStrokeColor} = props;
+
+  const allCaseData = dataConfiguration.joinedCaseDataArrays;
+  const allCircles = selectAllCircles(dotsElement); //includes both inner and outer circles
+  if (!allCircles) return;
   startAnimation(enableAnimation);
-  circles
-    .data(allCaseData, caseDataKeyFunc)
+
+  initializePoints({
+    selection: allCircles,
+    data: allCaseData,
+    pointRadius,
+    pointColor,
+    pointStrokeColor,
+    getPointColorAtIndex: (index) => pointColor,
+    instanceId
+  });
+  //----------------- Click on Dot -------------------
+  // dotsElement && select(dotsElement).on('click',
+  //   (event: MouseEvent) => {
+  //     const target = select(event.target as SVGSVGElement);
+  //     if (target.node()?.nodeName === 'circle') {
+  //       handleClickOnDot(event, (target.datum() as CaseData).caseID, dataConfiguration.dataset);
+  //     }
+  //   });
+  // dataConfiguration.setPointsNeedUpdating(false);
+}
+
+
+interface IInitializePoints {
+  selection: DotSelection;
+  data: any;
+  pointRadius: number;
+  pointColor: string;
+  pointStrokeColor: string;
+  getPointColorAtIndex?: (index: number) => string;
+  instanceId?: string;
+}
+
+const initializePoints = ({ selection, data, pointRadius,  pointColor,
+                            pointStrokeColor, getPointColorAtIndex, instanceId }: IInitializePoints) => {
+
+  //initialize outer highlight dots (this must be before the inner dots so that the inner dots are on top)
+  selection
+    .data(data)
+    .join(
+      (enter)=>
+        enter.append('circle')
+        .attr('class', 'graph-dot-highlighted')
+    );
+  //initialize inner dots
+  selection
+    .data(data)
     .join(
       (enter) =>
         enter.append('circle')
           .attr('class', 'graph-dot')
           .property('id', (d: CaseData) => `${instanceId}_${d.caseID}`),
-      (update) =>
-        update.attr('r', pointRadius)
-          .style('fill', pointColor)
-          .style('stroke', pointStrokeColor)
-          .style('stroke-width', defaultStrokeWidth)
     );
-  dotsElement && select(dotsElement).on('click',
-    (event: MouseEvent) => {
-      const target = select(event.target as SVGSVGElement);
-      if (target.node()?.nodeName === 'circle') {
-        handleClickOnDot(event, (target.datum() as CaseData).caseID, dataConfiguration.dataset);
+};
+
+export interface ISetPointSelection {
+  dotsRef: IDotsRef
+  dataConfiguration: IDataConfigurationModel
+  pointRadius: number,
+  selectedPointRadius: number,
+  pointColor: string,
+  pointStrokeColor: string,
+  getPointColorAtIndex?: (index: number) => string
+}
+
+
+export function setPointSelection(props: ISetPointSelection) {
+  const { dotsRef, dataConfiguration } = props;
+  const dataset = dataConfiguration.dataset;
+  const allCircles = selectAllCircles(dotsRef.current); //includes both inner and outer circles
+  const outerCircles = selectDots(dotsRef.current, true);
```

Comment:
We discussed creating functions like `selectInnerCircles` and `selectOuterCircles`. You should do this. It would be much more clear than passing a boolean to determine which set of circles is returned.